supply/spec/uploader_spec.rb

```diff
@@ -203,22 +203,69 @@ def find_obbs
       end
     end
 
-    describe '#perform_upload' do
-      let(:client) { double('client') }
-      let(:config) { { apk: 'some/path/app.apk' } }
+    RSpec::Matchers.define(:same_localizedtext_as) do |expected_arr|
+      match do |actual_arr|
+        actual_arr.sort_by(&:language).zip(expected_arr.sort_by(&:language)).map do |actual_localization, expected_localization|
+          actual_localization.language == expected_localization.language && actual_localization.text == expected_localization.text
+        end.reduce(:&)
+      end
+    end
 
-      before do
-        Supply.config = config
-        allow(Supply::Client).to receive(:make_from_config).and_return(client)
-        allow(client).to receive(:upload_apk).with(config[:apk]).and_return(1) # newly uploaded version code
-        allow(client).to receive(:begin_edit).and_return(nil)
-        allow(client).to receive(:commit_current_edit!).and_return(nil)
+    shared_examples 'run supply to upload metadata' do
+      describe '#perform_upload' do
+        subject(:release) { OpenStruct.new({ version_codes: version_codes }) }
```

Comment:
Probably worth using a `double('release', version_codes: version_codes)` instead here? (If I remember the syntax for creating rspec doubles correctly)